File: hw/elbay/u3/hw25TemplateMethod/hw25TemplateMethod/js/controller/MealController.js
const PastaDish = require("../model/PastaDish");
const SaladDish = require("../model/SaladDish");

class MealController {
    constructor(view) {
        this.view = view;
    }

    preparePasta() {
        this.view.showMessage("\nPreparing pasta ...");
        const pasta = new PastaDish();
        pasta.wantsGarnish = () => {
            const answer = this.view.getUserInput("Would you like parmesan on your pasta (y/n)?");
            return answer.toLowerCase().startsWith("y");
        };
        pasta.prepareMeal();
    }

    prepareSalad() {
        this.view.showMessage("\nPreparing salad ...");
        const salad = new SaladDish();
        salad.wantsGarnish = () => {
            const answer = this.view.getUserInput("Would you like croutons on your salad (y/n)?");
            return answer.toLowerCase().startsWith("y");
        };
        salad.prepareMeal();
    }
}

module.exports = MealController;

File: hw/elbay/u3/hw25TemplateMethod/hw25TemplateMethod/java/src/main/java/controller/MealController.java
package controller;

import model.*;
import view.MealView;

public class MealController {
    private MealView view;

    public MealController(MealView view) {
        this.view = view;
    }

    public void preparePasta() {
        view.showMessage("\nPreparing pasta ...");
        MealPreparation pasta = new PastaDish() {
            @Override
            public boolean wantsGarnish() {
                String answer = view.getUserInput("Would you like parmesan on your pasta (y/n)?");
                return answer.toLowerCase().startsWith("y");
            }
        };
        pasta.prepareMeal();
    }

    public void prepareSalad() {
        view.showMessage("\nPreparing salad ...");
        MealPreparation salad = new SaladDish() {
            @Override
            public boolean wantsGarnish() {
                String answer = view.getUserInput("Would you like croutons on your salad (y/n)?");
                return answer.toLowerCase().startsWith("y");
            }
        };
        salad.prepareMeal();
    }
}
